product_attribute_model_link/models/product_attribute.py

```diff
@@ -0,0 +1,337 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+def make_create():
+    """Instantiate a create method that generates new product attribute values
+    based on the current record."""
+
+    @api.model_create_multi
+    def create(self, vals_list, **kw):
+        product_attributes = self.env["product.attribute"].search(
+            [("linked_model_id", "=", self._name)]
+        )
+        if not product_attributes or self.env.context.get(
+            "operation_from_attribute_value"
+        ):
+            return create.origin(self, vals_list, **kw)
+        # Call original method
+        records = create.origin(self, vals_list, **kw)
```

Comment:
```python
# Call original method
records = create.origin(self, vals_list, **kw)
if not product_attributes or self.env.context.get("operation_from_attribute_value"):
    return records
        
``` 

To avoid code duplication, you can write like this